### PR TITLE
Add CBOR encoded Datum and RedeemerData

### PR DESCRIPTION
**Pull request by sorki**
Forward port of #1110

### CODE REVIEW

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx/Alonzo.hs [hunk 1]
```diff
@@ -293,18 +293,24 @@ getPlutusScriptSize script =
...
 
 mkTxData :: (Ledger.DataHash StandardCrypto, Alonzo.Data era) -> PlutusData
-mkTxData (dataHash, dt) = PlutusData (dataHashToBytes dataHash) (encodeData dt)
+mkTxData (dataHash, dt) = PlutusData (dataHashToBytes dataHash) (jsonData dt) (cborData dt)
+  where
+    jsonData :: Alonzo.Data era -> ByteString
+    jsonData =
+          LBS.toStrict
+        . Aeson.encode
+        . Api.scriptDataToJson Api.ScriptDataJsonDetailedSchema
```
[the remark this answers is not in the document]
Comment (sorki) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx/Alonzo.hs:309]: I've just switched this one to point free style. Ad detailed vs simple schema - https://github.com/input-output-hk/cardano-db-sync/pull/807#discussion_r718494540

**Comment by sorki** [comment]
I've dropped the `lazy insert` commits since #1134 has all that.

**Comment by kderme** [comment]
This has been merged to a different branch insetad of master. @sorki would you mind retargeting master and reopening a new pr?

**Comment by sorki** [comment]
> This has been merged to a different branch insetad of master. @sorki would you mind retargeting master and reopening a new pr?

Sure, created #1142

**Pull request merged**